### PR TITLE
API-1835: realign manifestclient on fs.FS

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
/lgtm

/hold

so that @bertinatto can take a look as well.

File: pkg/manifestclienttest/client_write_test.go [hunk 1]
```diff
@@ -207,6 +212,7 @@ func TestSimpleWritesChecks(t *testing.T) {
 
 					if !manifestclient.AreAllSerializedRequestsEquivalent(actualRequests, expectedRequests.AllRequests()) {
 						t.Logf("Re-run with `UPDATE_MUTATION_TEST_DATA=true` to write new expected test data")
+						t.Log(expectedRequests.AllRequests())
```
Comment (p0lyn0mial) [inline pkg/manifestclienttest/client_write_test.go:215]: is this helpful ?

**Review by bertinatto — COMMENTED** [review]
/lgtm

File: pkg/manifestclienttest/client_write_test.go (hunk 1 above)
Comment (deads2k) [reply pkg/manifestclienttest/client_write_test.go:215]: Yeah, helped find a bug.

**Pull request merged**